autotest.pm

```diff
@@ -217,7 +217,7 @@ sub run_all {
         $died = 1;    # test execution died
     }
     eval {
-        bmwqemu::save_vars();
+        bmwqemu::save_vars(no_secret => 1);
```

Comment:
it is not the only place where `save_vars` actually called. Did you check others ? For example `save_vars` called on each `set_var` call so looks like your secrets will still leak after this  PR. 

Generally I don't see why we need to have another flavor of `save_vars` which actually saving them

Reply:
@asmorodskyi It need to be the last part where the save_vars method is called, as this is the one that saves the final vars.json file that will be updated in the end to the webUI.

Reply:
@asmorodskyi Good point. Besides, would it ever make sense to call this with `no_scret => 0`? If not, I would not make it an option at all.

Reply:
the problem is that we use vars.json dump and save to transfer it between backend and autotest processes.

Reply:
Yes. If I would always avoid writing of such variables, the first test-module sees them, but not the second one during one job.